library/alloc/tests/vec.rs

```diff
@@ -1107,8 +1109,31 @@ fn test_into_iter_drop_allocator() {
 
 #[test]
 fn test_into_iter_zst() {
-    for _ in vec![[0u64; 0]].into_iter() {}
-    for _ in vec![[0u64; 0]; 5].into_iter().rev() {}
+    #[derive(Debug, Clone)]
+    struct AlignedZstWithDrop([u64; 0]);
+    impl Drop for AlignedZstWithDrop {
+        fn drop(&mut self) {
+            let addr = self as *mut _ as usize;
+            assert!(hint::black_box(addr) % mem::align_of::<u64>() == 0);
```

Comment:
Seems reasonable enough. Good thing we didn't call it `hint::bench_black_box` :p

Reply:
Yeah I guess it is useful for some opportunistic UB-detection tests as well. ;)